File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.java
class Solution {
    public int solution(String s) {
        int answer = 0;
        int different = 0;
        int origin = 0;
        boolean isStart = false;
        char ch = '!';
        for(int i = 0;i<s.length();i++){
            if(!isStart){
                isStart = true;
                ch = s.charAt(i);
                origin++;
                continue;
            }
            if(isStart){
                if(s.charAt(i)==ch) origin++;
                else different++;
            }
            if(different==origin){
                answer++;
                different = 0;
                origin = 0;
                isStart= false;
            }
        }
        if(origin!=0 || different!=0) answer++;
        return answer;
    }
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    first = 0
    second = 0
    fIndex = -1
    
    for i in range(0,len(s)) :
        if fIndex  == -1 :
            fIndex = i
            first = 1
            continue
        else :
            if s[i] == s[fIndex] :
                first+=1
            else :
                second+=1
        if first == second :
            answer+=1
            fIndex = -1
            first = 0
            second = 0
    if fIndex != -1:
        answer += 1    
    return answer